pkg/sql/catalog/descs/descriptor.go

```diff
@@ -398,139 +518,13 @@ func (tc *Collection) finalizeDescriptors(
-		descs[i], err = tc.hydrateTypesInDescWithOptions(ctx, txn, hd, flags.IncludeOffline, flags.AvoidLeased)
-		if err != nil {
-			return err
-		}
-	}
-	return nil
-}
-
 func (tc *Collection) deadlineHolder(txn *kv.Txn) deadlineHolder {
 	if tc.maxTimestampBoundDeadlineHolder.maxTimestampBound.IsEmpty() {
 		return txn
 	}
 	return &tc.maxTimestampBoundDeadlineHolder
 }
 
-// Getting a schema by name uses a special resolution path which can avoid
-// a namespace lookup because the mapping of database to schema is stored on
-// the database itself. This is an important optimization in the case when
-// the schema does not exist.
-//
-// TODO(ajwerner): Understand and rationalize the namespace lookup given the
-// schema lookup by ID path only returns descriptors owned by this session.
-//
-// The alwaysLookupLeasedPublicSchema parameter indicates that a missing public
-// schema entry in the database descriptor should not be interpreted to
-// mean that the public schema is the synthetic public schema, and, instead
-// the public schema should be looked up via the lease manager by name.
-// This is a workaround activated during the public schema migration to
-// avoid a situation where the database does not know about the new public
-// schema but the table in the lease manager does.
-//
-// TODO(ajwerner): Remove alwaysLookupLeasedPublicSchema in 22.2.
-func getSchemaByName(
-	ctx context.Context,
-	tc *Collection,
-	txn *kv.Txn,
-	db catalog.DatabaseDescriptor,
-	name string,
-	avoidLeased, mutable, avoidSynthetic bool,
-	alwaysLookupLeasedPublicSchema bool,
-) (bool, catalog.Descriptor, error) {
-	if !db.HasPublicSchemaWithDescriptor() && name == tree.PublicSchema {
-		// TODO(ajwerner): Remove alwaysLookupLeasedPublicSchema in 22.2.
-		if alwaysLookupLeasedPublicSchema {
-			desc, _, err := tc.leased.getByName(ctx, txn, db.GetID(), 0, catconstants.PublicSchemaName)
-			if err != nil {
-				return false, desc, err
-			}
-			return true, desc, nil
-		}
-		return true, schemadesc.GetPublicSchema(), nil
-	}
-	if sc := tc.virtual.getSchemaByName(name); sc != nil {
-		return true, sc, nil
-	}
-	if isTemporarySchema(name) {
-		if isDone, sc := tc.temporary.getSchemaByName(ctx, db.GetID(), name); sc != nil || isDone {
-			return sc != nil, sc, nil
-		}
-		scID, err := tc.stored.LookupDescriptorID(ctx, txn, db.GetID(), keys.RootNamespaceID, name)
-		if err != nil || scID == descpb.InvalidID {
-			return false, nil, err
-		}
-		return true, schemadesc.NewTemporarySchema(name, scID, db.GetID()), nil
-	}
-	if id := db.GetSchemaID(name); id != descpb.InvalidID {
-		// TODO(ajwerner): Fill in flags here or, more likely, get rid of
-		// it on this path.
-		sc, err := tc.getSchemaByID(ctx, txn, id, tree.SchemaLookupFlags{
-			RequireMutable: mutable,
-			AvoidLeased:    avoidLeased,
-			AvoidSynthetic: avoidSynthetic,
-		})
-		if errors.Is(err, catalog.ErrDescriptorDropped) {
-			err = nil
-		}
-		return sc != nil, sc, err
-	}
-	return false, nil, nil
-}
```

Comment:
The special schema-lookup-path has been unified with the others now. It wasn't terribly different except notably for an early exit if the schema can't possibly be anything but virtual.